### DISCUSSION TITLE
deduce groupid from previous //DEPS ?

### DISCUSSION THREAD

**Discussion by maxandersen** [jbangdev/jbang#1545] [Ideas]
just a crazy thought ...should we allow this:

```
//DEPS io.quarkus:quarkus-bom:${quarkus.version:2.15.3.Final}@pom
//DEPS quarkus-picocli
//DEPS resteasy-reactive-jackson
//DEPS rest-client-reactive-jackson
```

vs this:

```
//DEPS io.quarkus:quarkus-bom:${quarkus.version:2.15.3.Final}@pom
//DEPS io.quarkus:quarkus-picocli
//DEPS io.quarkus:resteasy-reactive-jackson
//DEPS io.quarkus:rest-client-reactive-jackson
```

basically if you are specifying //DEPS's then if no GroupID present assume you meant the last one used in the same file?

what could go wrong?

**Comment by fbricon** [comment]
it will make tools job harder, unless JBang provides a map between dependency declaration (line, offset) and actual GAVs.

**Reply by maxandersen** [reply]
Beyond completion where would you need to use that mapping ?

**Reply by maxandersen** [reply]
An would a artifact to GAV be sufficient ? Ie. List of what the string in file to what gav it resulted in?

**Comment by quintesse** [comment]
Honestly the only reason I would find shortcuts like this interesting is if I would often be manually typing in the artifact names myself, but who does that? 🙂 
I would either be using the auto-complete in the IDE or I would be copying and pasting GAVs from somewhere.
